gateway/ln-gateway/src/rpc/mod.rs

```diff
@@ -110,10 +112,23 @@ pub struct WithdrawPayload {
     pub address: Address,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct FederationInfo {
     pub federation_id: FederationId,
     pub mint_pubkey: XOnlyPublicKey,
+    pub balance: Option<Amount>,
+}
```

Comment:
@okjodom I got stuck on a better way for this, do you have anything in mind ?

I was thinking that using something like [serde_with](https://docs.rs/serde_with/2.3.1/serde_with/attr.skip_serializing_none.html) would help to keep a single struct, but I'm not sure if adding a new dependency would be a bad idea 🤔

Reply:
**Describing the scenario to rebuild context**

- When we use `/get_info` API without a filter, we need to return `GatewayInfo` struct which lists partials of `FederationInfo` structs.
- When we use `/get_info` API has a filter, i.e. `/get_info <federation_id>`, we need to return  `GatewayInfo` struct which potentially lists  a single, NON-partial `FederationInfo`.

Making `Option`al fields in `FederationInfo` would my go to for making partials `FederationInfo`. Are there alternative patterns? @Maan2003, @dpc

Reply:
![image](https://user-images.githubusercontent.com/11217077/229469875-0057d814-a7b2-40c8-a011-85059c51d600.png)

Reply:
use `serde(flatten)`

```rust
struct Partial { ... };

struct Complete {
   #[serde(flatten)]
   partial: Partial,
   ....
};

Reply:
or the other way around

```rust
struct Foo {
  // requried_fields ...
  complete: Option<Complete>;
};

struct Complete { ... };
```

Reply:
@okjodom @Maan2003 I was trying this approach but I'm still not sure how the GatewayInfo would be able to handle both `PartialFederationInfo` and `FederationInfo`, as they would be two different types 🤔.

Should the Partial pattern handle that, am I missing something ?

[I've tried to solve it by using a `FederationInfo` enum alongside the Partial structures](https://github.com/fedimint/fedimint/pull/2046/commits/63275fafbbae740935f01304cdf88ce9cefe1360), but it doesn't seem to help.

Reply:
@okjodom @Maan2003 fyi, trying to apply the Partial pattern was getting really complicated.

Taking a deeper look at the `serde` docs I've found the skip... attributes, and I've used the [`skip_serializing_if`](https://serde.rs/field-attrs.html#skip_serializing_if) as is really suitable for the expected behavior, but please let me know what you think about it.